hikari/utilities/iterators.py

```diff
@@ -797,7 +833,30 @@ async def __anext__(self) -> ValueT:
         async for item in self._iterator:
             if self._predicate(item):
                 return item
-        raise StopAsyncIteration
+
+        self._complete()
+
+
+class _ChunkedLazyIterator(typing.Generic[ValueT], LazyIterator[typing.Sequence[ValueT]]):
+    __slots__: typing.Sequence[str] = ("_iterator", "_chunk_size")
+
+    def __init__(self, iterator: LazyIterator[ValueT], chunk_size: int) -> None:
+        self._iterator = iterator
+        self._chunk_size = chunk_size
+
+    async def __anext__(self) -> typing.Sequence[ValueT]:
+        chunk = []
+
+        async for item in self._iterator:
+            chunk.append(item)
+
+            if len(chunk) == self._chunk_size:
```

Comment:
could this just be 
```py
return await self.limit(chunk_size)
```

Reply:
wouldn't that introduce extra over head?

Reply:
not overly but if you prefer it like this then go for it.